Automattic/wp-calypso
# Post Revisions: Move UI to a modal Dialog (from Editor Sidebar)

This takes the Post Revisions UI out of the Editor Sidebar & into a `Dialog` activated by the `History` button.

<img width="1327" alt="screen shot 2017-11-05 at 10 12 33 pm" src="https://user-images.githubusercontent.com/1587282/32423783-97fb6486-c276-11e7-9e36-10806c8c22be.png">

Some redux state was added and enhanced to make this possible.

## To Test

* in dev env: Open a post in the editor
* `History` button should appear as before
* Clicking `History` should open a modal dialog w/ a nice layout
* Clicking a "revision" should show the diff on the right
* <kbd>j</kbd> & <kbd>k</kbd> hotkeys should navigate through a posts revision history (h/t #19202)
* Selecting a revision & clicking `Load` should put the selected revision's title and content into the editor inputs.
* Saving after loading a revision should be sane & predictable.

### Test behind the flag

* Append the following query arg to the dev URL to load without the feature flag: `?flags=-post-editor/revisions`
* The `History` button should not appear in the Ground Control area
* There should be a revisions count in the editor's `Post Settings | Options` sidebar.
  * Clicking it should take you to the post revisions feature in wp-admin in a new tab

### Follow up work
- Improve loading placeholder ([1](https://github.com/Automattic/wp-calypso/pull/19197#discussion_r150632778))
- clean up of redundant nested sidebar 
- remove 'is_shown' class from `post-editor__inner-content` - it was added for the original implementation of revisions to allow for hiding this element and is now redundant.
- support for arbitrary revisions ([1](https://github.com/Automattic/wp-calypso/pull/19197#discussion_r150631857))
- Incorporate help info as in the wp-admin Overview, or similar ([1](https://github.com/Automattic/wp-calypso/pull/19197#issuecomment-343779418))
- Investigate and fix extrenous modal content scrollbars & fix content element size.
- Decouple `EditorRevisionsDialog` from the history button (it should be openable from anywhere) ([1](https://github.com/Automattic/wp-calypso/pull/19197#discussion_r150354371)).
- Replace `toggleDialog` with explicit open & close methods.
- Investigate tabbing/focusing between diff and revisions list - could improve keyboard workflow & accessibility
- Reconcile styling & placement of History button with the "save..." button ([1](https://github.com/Automattic/wp-calypso/pull/19197#issuecomment-342124504))
- The word "minor" alone is not great for translations or for context in English. We should consider adding the "changes" suffix back, showing empty space, or using iconography instead
- Add times to Revision list items w/o relative times so it's easier to distinguish between them
